### ISSUE TITLE
Overlapping marker spiderfier bug on layers using custom markers

### ISSUE THREAD

**Issue by crisner** [publiclab/leaflet-environmental-layers#271]
**Issue:** The implementation of overlapping marker spiderfier plugin has an issue where the spider legs are located off of the markers and does not point to them correctly as can be seen below in the screenshot of the map with pfaslayer in use.
![oms-issue-before](https://user-images.githubusercontent.com/29401459/67163558-17b2a500-f38e-11e9-96a9-e56d1156bd16.png)

The problem seems to be because the values provided for the `iconAnchor` property, while creating the custom marker, are messing with the marker's origin. The screenshot below shows the problem resolved after removing the `iconAnchor` property from the icon class.
![oms-issue-after](https://user-images.githubusercontent.com/29401459/67163658-251c5f00-f38f-11e9-817e-4f86a76b296e.png)

**Solution:** The `iconAnchor` values has to be correctly set or removed to get the desired behavior from the plugin. In case the property has to be used as it is, then we could remove the use of the plugin for that layer.

**Comment by crisner** [comment]
@sagarpreet-chadha Wanted to list the issue so that we can decide on how to proceed with the issue now or at a later time. Depending on how we plan on implementing the solution this issue can be listed for the first-timers to contribute to this project.

**Comment by YuetChan** [comment]
Hi I I would like to work on this issue. Do I just simply remove the use of the plugin for pfaslayer layer? Thank you

**Comment by ajitsinghkaler** [comment]
ypua re working on it

**Comment by crisner** [comment]
Thank you @YuetChan ! @ajitsinghkaler  I have asked @cesswairimu to assign another issue to you as @YuetChan had commented on this first.

**Comment by YuetChan** [comment]
@crisner For this issue, do I remove the use of the plugin for pfaslayer by omitting oms.addMarker(marker) in pfaslayer.js file?

**Comment by crisner** [comment]
Change the code below:
https://github.com/publiclab/leaflet-environmental-layers/blob/0b5b0bc9ff50a85113854efd2761724e770196cb/src/pfasLayer.js#L1-L8
to:
```javascript
L.Icon.PfasLayerIcon = L.Icon.extend({
   options: {
    iconUrl: 'https://openclipart.org/image/300px/svg_to_png/117253/1297044906.png',
    iconSize:     [10, 10],
    popupAnchor:  [1, -2]
  }
});
```
You will need to remove the iconAnchor property and change the values in the popupAnchor property.

**Comment by ajitsinghkaler** [comment]
@crisner If @YuetChan is not working on it I have already fixed it. Should I create a pull request for it?

**Comment by crisner** [comment]
No. You don't have to add the changes made by grunt build.

**Comment by crisner** [comment]
@ajitsinghkaler if you are interested you could work on the odorlog layer which has the same issue. You may need to experiment with different values though so that the icon and the popup render in their correct positions.

**Comment by ajitsinghkaler** [comment]
Okay, shall I create a new Issue for it?

**Comment by ananyaarun** [comment]
This issue in LEL is reserved for GCI students.
Thanks !!

**Comment by jywarren** [comment]
Ah, i'm sorry, this was already solved in #277 and #275! @ananyaarun is that ok? Thanks, everyone, for your excellent help!

**Comment by ananyaarun** [comment]
Hmm yes @jywarren .
Lets just remove the GCI label from this then.
